crates/bevy_sprite/src/tilemap_chunk/mod.rs

```diff
@@ -0,0 +1,176 @@
+use crate::MeshMaterial2d;
+use bevy_app::{App, Plugin, Update};
+use bevy_asset::{Assets, Handle, RenderAssetUsages};
+use bevy_derive::{Deref, DerefMut};
+use bevy_ecs::{
+    component::Component,
+    entity::Entity,
+    observer::Trigger,
+    query::Changed,
+    resource::Resource,
+    system::{Commands, Query, ResMut},
+    world::OnAdd,
+};
+use bevy_image::Image;
+use bevy_math::{primitives::Rectangle, UVec2};
+use bevy_platform::collections::HashMap;
+use bevy_render::{
+    mesh::{Mesh, Mesh2d},
+    render_resource::{Extent3d, TextureDimension, TextureFormat},
+};
+use tracing::warn;
+
+mod tilemap_chunk_material;
+
+pub use tilemap_chunk_material::*;
+
+/// Plugin that handles the initialization and updating of tilemap chunks.
+/// Adds systems for processing newly added tilemap chunks and updating their indices.
+pub struct TilemapChunkPlugin;
+
+impl Plugin for TilemapChunkPlugin {
+    fn build(&self, app: &mut App) {
+        app.init_resource::<TilemapChunkMeshMap>()
+            .add_observer(on_add_tilemap_chunk)
+            .add_systems(Update, update_tilemap_chunk_indices);
+    }
+}
+
+/// A resource storing the meshes for each tilemap chunk size.
+#[derive(Resource, Default, Deref, DerefMut)]
+struct TilemapChunkMeshMap(HashMap<UVec2, Handle<Mesh>>);
```

Comment:
```suggestion
pub struct TilemapChunkMeshMap(HashMap<UVec2, Handle<Mesh>>);
```

Just in case a third party crate needs access to it.